test/parameter.js

```diff
@@ -177,4 +120,55 @@ describe('Parameter to Member converter', () => {
     expect(enumerations).to.be.instanceof(minim.elements.Array);
     expect(enumerations.toValue()).to.deep.equal([['hello']]);
   });
+
+  it('creates a warning when example does not match parameter type', () => {
+    const parser = new Parser({ minim, source: '' });
+    parser.result = new minim.elements.ParseResult();
+    parser.convertParameterToMember({
+      type: 'string',
+      'x-example': 5,
```

Comment:
Previously they were not coerced as strings. It would become an invalid element, for example an array element that had a content of string. For the following example:

```yaml
type: array
items:
  type: number
x-example: hello
```

I don't think we can do much here other than drop the `hello` it is not an array and it is certainly not a number to go inside an array. The user was given a warning and they should know how to fix the problem.